### DISCUSSION TITLE
BigInt vs number

### DISCUSSION THREAD

**Discussion by MauriceArikoglu** [prisma/prisma#6948] [Q&A] [answered]
I want to use JS `number`, but need `int8` as db row type. To work with `int8` I need to assign `BigInt` as Prisma type within schema.prisma, which then in turn kills compatibility with `number`.

Can I specify `Int` as type in schema.prisma and annotate it somehow so the underlying db uses `int8` instead of `int4`?

**Comment by elie222** [comment]
Looking to do the same thing. We have a number like `2233459262` which is fine in JS but when stored in prisma returns a negative value as it's larger than `int4`. We want to use BigInt for this number but we don't want JS treating the field as a BigNumber, but regular number instead.

**Comment by austincrim** [comment, marked as answer]
Hey @MauriceArikoglu 👋 !

We have an existing feature request for this: #7570. Would you mind adding your +1 or specific use case to help us prioritize this?

Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
